### PR TITLE
F/content metadata

### PR DESCRIPTION
**Pull request by tomwayson**
introduces `getContenMetadata()` that fetches item metadata and parses it as JSON.

@mikeringrose as mentioned in https://github.com/Esri/hub.js/issues/353#issuecomment-689826285, this uses same library (although a more recent version) and equivalent configuration as TheDuke does to parse the metadata as JSON. The major difference is that `getContentMetadata()` uses `@esri/arcgis-rest-portal` under the hood to fetch the metadata from portal.  The long term hope is that we could use this code in TheDuke (someday). Can you please look for anything that would prohibit that.

### CODE REVIEW

File: packages/content/src/metadata.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import { getItemMetadata } from "@esri/arcgis-rest-portal";
+import { IHubRequestOptions } from "@esri/hub-common";
+import { parse } from "fast-xml-parser";
```
Comment (drewctate) [inline packages/content/src/metadata.ts:3]: Is this what we use in the indexer @mikeringrose ?

**Review by drewctate — APPROVED** [review]
LGTM

File: packages/content/src/metadata.ts (hunk 1 above)
Comment (tomwayson) [reply packages/content/src/metadata.ts:3]: it is, though this PR is using v3.x, but indexer uses 2.x, and there are a few differences in the configuration settings and default behavior between the versions. I've made the equivalent configuration in this PR, but there are still slight differences in how the XML is parsed. For example, the [`peXml` node](https://github.com/Esri/hub.js/pull/375/files#diff-720828cca6aaaac8de88dda7278ca80fR35) is not parsed by v2 b/c it contains a nested XML document, but v3 parses that XML in place. See https://github.com/Esri/hub.js/pull/375/files#diff-720828cca6aaaac8de88dda7278ca80fR317-R324

Comment (mikeringrose) [reply packages/content/src/metadata.ts:3]: This is what we use. I wish we would have turned to it into nicer json, but it is what it is.

**Pull request merged**